### PR TITLE
use url hash to select tab

### PR DESCRIPTION
**Pull request by MauserBitfly**
This PR:
- introduces a new composable to select tabs based on the url hash
- adds this composable to the dashboard page

Note to the tester: url hashes are not sent to the server and therefor not available during SSR

### CODE REVIEW

File: frontend/composables/useHashTabs.ts
Comment (D13ce) [inline frontend/composables/useHashTabs.ts:1]: **Bug:** I am on a private dashboard and click the rewards tab to get the `#rewards'' in the URL. I then head-reload (`CTRL`+`F5`) and get the following two warnings:
image
```
[nuxt] Calling `useRoute` within middleware may lead to misleading results. Instead, use the (to, from) arguments passed to the middleware to access the new and old routes. 6 [composables:12:13](https://local.beaconcha.in:3000/_nuxt/node_modules/nuxt/dist/app/composables/)
[Vue Router warn]: Couldn't find element using selector "#rewards" returned by scrollBehavior. 2 [vue-router.mjs:35:17](https://local.beaconcha.in:3000/_nuxt/node_modules/vue-router/dist/vue-router.mjs)
```

**Review by D13ce — CHANGES_REQUESTED** [review]
Nice and future proof implementation, well done. Just spotted 2 warnings, see my comment below :point_down:

File: frontend/composables/useHashTabs.ts
Comment (MauserBitfly) [reply frontend/composables/useHashTabs.ts:1]: fixed these two warnings

Comment (D13ce) [reply frontend/composables/useHashTabs.ts:1]: Switching tabs now causes the page to scroll to the top.
Furthermore, reloading while being on `#rewards'' causes the page to switch to `#summary`.

Comment (MauserBitfly) [reply frontend/composables/useHashTabs.ts:1]: oh sorry, should work again

**Review by D13ce — APPROVED** [review]
LGTM

**Pull request merged**